lib/ansible/plugins/strategy/linear.py

```diff
@@ -200,7 +200,7 @@ def run(self, iterator, play_context):
                         results.extend(self._execute_meta(task, play_context, iterator, host))
                         if task.args.get('_raw_params', None) not in ('noop', 'reset_connection', 'end_host', 'role_complete', 'flush_handlers'):
                             run_once = True
-                        if (task.any_errors_fatal or run_once) and not task.ignore_errors:
+                        if task.any_errors_fatal and not task.ignore_errors:
```

Comment:
I am not sure this is correct. There are several things at play here. Meta tasks' results do not go through the normal results processing in `StrategyBase._process_pending_results()` (until possibly https://github.com/ansible/ansible/pull/83252 is merged anyway) which contains the code to handle the `run_once` functionality (if the task fails and not `ignore_errors`, fail all other hosts) so that means that `run_once` is not implemented for `meta`. I think that is why the local variable `any_errors_fatal` is set also for `run_once` so the code that handles `any_errors_fatal` is used to "implement" `run_once` for `meta` tasks because effectively both features just fails all other hosts.

But even if the meta tasks' results went through the normal results processing, it still would not work because there we check `original_task.run_once` to trigger the `run_once` functionality but that value does not reflect what we want because above we just set `run_once` to a local variable here for hardcoded set of meta tasks.

Reply:
On the other hand, `StrategyBase._execute_meta()` does not appear to create a failing `TaskResult` ever. That would mean that the `any_errors_fatal` code would not be triggered because for that it requires a failing task result and therefore `run_once` on meta task only means that the task would be run on only one host and the second part of the feature (failing other hosts in case of a task failure) is not needed? So this change is in the end fine??